pkg/controller/configmapcainjector/starter/starter.go

```diff
@@ -2,49 +2,43 @@ package starter
 
 import (
 	"crypto/x509"
-	"encoding/pem"
 	"fmt"
 	"io/ioutil"
 	"time"
 
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/informers"
 	"k8s.io/client-go/kubernetes"
-	"k8s.io/client-go/rest"
 
-	servicecertsignerv1alpha1 "github.com/openshift/api/servicecertsigner/v1alpha1"
+	scsv1alpha1 "github.com/openshift/api/servicecertsigner/v1alpha1"
 	"github.com/openshift/library-go/pkg/controller/controllercmd"
 	"github.com/openshift/service-ca-operator/pkg/controller/configmapcainjector/controller"
 )
 
-func ToStartFunc(config *servicecertsignerv1alpha1.ConfigMapCABundleInjectorConfig) (controllercmd.StartFunc, error) {
+func StartConfigMapCABundleInjector(ctx *controllercmd.ControllerContext) error {
+	config := &scsv1alpha1.ConfigMapCABundleInjectorConfig{}
+	if ctx.ComponentConfig != nil {
+		// make a copy we can mutate
+		configCopy := ctx.ComponentConfig.DeepCopy()
+		// force the config to our version to read it
+		configCopy.SetGroupVersionKind(scsv1alpha1.GroupVersion.WithKind("ConfigMapCABundleInjectorConfig"))
+		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(configCopy.Object, config); err != nil {
+			return err
+		}
+	}
 	if len(config.CABundleFile) == 0 {
-		return nil, fmt.Errorf("no ca bundle provided")
+		return fmt.Errorf("no ca bundle provided")
 	}
-
-	ca, err := ioutil.ReadFile(config.CABundleFile)
+	pemBlock, err := ioutil.ReadFile(config.CABundleFile)
 	if err != nil {
-		return nil, err
-	}
-
-	// Verify that there is at least one cert in the bundle file
-	block, _ := pem.Decode(ca)
-	if block == nil {
-		return nil, fmt.Errorf("failed to parse CA bundle file as pem")
+		return err
 	}
-	if _, err = x509.ParseCertificate(block.Bytes); err != nil {
-		return nil, err
+	if !x509.NewCertPool().AppendCertsFromPEM(pemBlock) {
+		return fmt.Errorf("failed to find CA bundle in file %s", config.CABundleFile)
```

Comment:
Please keep the original error as well, while there's probably very little that can go wrong while trying to parse PEM block, I'd still say there would be cases where having the original error could prove helpful.

Reply:
changed back to original in 16f36c01